### PR TITLE
fix(Microsoft SQL Node): Prevent SQL injection

### PR DESCRIPTION
**Pull request by michael-radency**
Github issue / Community forum post (link here to close automatically):

### CODE REVIEW

File: packages/nodes-base/nodes/Microsoft/Sql/MicrosoftSql.node.ts [hunk 1]
```diff
@@ -327,10 +324,25 @@ export class MicrosoftSql implements INodeType {
...
+								for (const [vIndex, value] of row.entries()) {
+									request.input(`r${rIndex}v${vIndex}`, value);
+								}
+							}
+
+							const query = `INSERT INTO [${table}] (${formatColumns(
+								columnString,
+							)}) VALUES ${valuesPlaceholder.join(', ')};`;
+
+							return request.query(query);
```
[the remark this answers is not in the document]
Comment (netroy) [reply packages/nodes-base/nodes/Microsoft/Sql/MicrosoftSql.node.ts:345]: nevermind. just learn about the `.input` method used above.

**Review by netroy — COMMENTED** [review]
Can we please add unit tests with mocked `mssql` module to assert that the calls to `.input` and `.query` are what we expect them to be?

File: packages/nodes-base/nodes/Microsoft/Sql/MicrosoftSql.node.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (michael-radency) [reply packages/nodes-base/nodes/Microsoft/Sql/MicrosoftSql.node.ts:345]: we could probably get rid of those, still WIP
```
	return Object.values(item).map((val) => {
		//the column cannot be found in the input
		//so, set it to null in the sql query
		if (val === null) {
			return 'NULL';
		} else if (typeof val === 'boolean') {
			return +!!val;
		}
		return val;
	});
```

File: packages/nodes-base/nodes/Microsoft/Sql/test/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+	configurePool,
+	deleteOperation,
+	insertOperation,
+	updateOperation,
+} from '../GenericFunctions';
+
+jest.mock('mssql', () => {
+	const originalModule = jest.requireActual('mssql');
+	const { Request } = originalModule;
+	Request.prototype.query = jest.fn(async function () {
```
Comment (netroy) [inline packages/nodes-base/nodes/Microsoft/Sql/test/utils.test.ts:12]: can we please also mock `.input` calls, and asserts that the values are correct s well, and not just the templated queries 🙏🏽 ?

**Review by netroy — CHANGES_REQUESTED** [review]
Haven't actually tested this, because I don't have mssql setup. 
but, besides the one comment, LGTM.

**Comment by michael-radency** [comment]
> Haven't actually tested this, because I don't have mssql setup. but, besides the one comment, LGTM.

setup is rather easy with docker, here's is guide:
https://learn.microsoft.com/en-us/sql/linux/quickstart-install-connect-docker?view=sql-server-ver16&pivots=cs1-bash

File: packages/nodes-base/nodes/Microsoft/Sql/test/utils.test.ts (hunk 2 above)
Comment (michael-radency) [reply packages/nodes-base/nodes/Microsoft/Sql/test/utils.test.ts]: @netroy 
updated tests with values checks

**Review by netroy — APPROVED** [review]
👍🏽

**Comment by janober** [comment]
Got released with [`n8n@1.14.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.14.0)

**Pull request merged**